src/compiler/crystal/tools/dependencies.cr

```diff
@@ -0,0 +1,60 @@
+require "set"
+require "colorize"
+require "../syntax/ast"
+
+class Crystal::Command
+  private def dependencies
+    dependency_printer = nil
+    option_parser = parse_with_crystal_opts do |opts|
+      opts.on("-f FORMAT", "--format FORMAT", "Format the output. Available options: flat, tree (default)") do |format|
+        case format
+        when "flat"
+          dependency_printer = DependencyPrinter.new(STDOUT, flat: true)
+        when "tree"
+          dependency_printer = DependencyPrinter.new(STDOUT, flat: false)
+        else
+          abort "Invalid format: #{format}"
+        end
+      end
+    end
+
+    config = create_compiler "tool dependencies", no_codegen: true
+    config.compiler.no_codegen = true
+
+    config.compiler.dependency_printer = dependency_printer || DependencyPrinter.new(STDOUT)
+    config.compile
```

Comment:
Can this be expressed as an extra parameter of `Crystal::Command#compile_no_codegen`? In that case the command-line argument handling would be done inside `#create_compiler` itself, similar to `hierarchy`

Reply:
What would be the benefit? It seems to me this is a pretty simple setup. Yes, it's duplicate code. But combining everthing in one place adds quite a bit of complexity.

Reply:
I haven't run this branch yet, but I believe this means the tool doesn't have access to any of the common command-line options that other tools have (`-D`, `--prelude` etc.)

Reply:
There is indeed a problem with this approach: `crystal tool dependencies` still shows the full help rather than this help, but the option parser only ever accepts this `-f`, and not the other options like `-h`. In particular this `-f tree|flat` conflicts with the other `-f text|json`.

Reply:
If `crystal tool hierarchy` accepts `-D` and `--prelude`, so should this tool; this is not merely a matter of code duplication

Reply:
The actual common piece that should be reused for a single option parser instance is `create_compiler`. Utilizing `compile_no_codegen` would add unnecessary complexity for no real gain.